components/collapsible-panel/collapsible-panel-group.js

```diff
@@ -32,39 +27,32 @@ class CollapsiblePanelGroup extends SkeletonGroupMixin(LitElement) {
 
 	constructor() {
 		super();
-		this._panels = [];
+		this._panels = new SubscriberRegistryController(this, 'collapsible-panel-group');
 	}
 
 	render() {
 		const classes = {
 			spaced: this._panels?.[0]?.type !== 'inline',
 		};
 
-		return html`<slot class="${classMap(classes)}" @slotchange="${this._handleSlotChange}"></slot>`;
-	}
-
-	async _getPanels() {
-		const slot = this.shadowRoot?.querySelector('slot');
-		if (!slot) return;
-
-		return slot.assignedNodes({ flatten: true }).filter((node) => node.nodeType === Node.ELEMENT_NODE);
-	}
-
-	async _handleSlotChange() {
-		this._panels = await this._getPanels();
-		this._updatePanelAttributes();
+		return html`<slot class="${classMap(classes)}"></slot>`;
 	}
 
 	_updatePanelAttributes() {
-		if (!this._panels || this._panels.length === 0) return;
-
-		if (this._panels[0]?.type !== 'inline') return;
-
-		for (const panel of this._panels) {
-			panel._noBottomBorder = true;
+		const panels = this.shadowRoot
+			?.querySelector('slot')
+			?.assignedNodes({ flatten: true })
+			.filter((node) => node.nodeType === Node.ELEMENT_NODE && node.tagName === 'D2L-COLLAPSIBLE-PANEL');
+		if (!panels || panels.length === 0) return;
+
+		const isInline = panels[0].type === 'inline';
+		for (let i = 0; i < panels.length; i++) {
+			if (i < panels.length - 1) {
+				panels[i]._noBottomBorder = isInline;
+			} else {
+				panels[i]._noBottomBorder = false;
+			}
```

Comment:
I believe this has the same result. Shorter, skips running if not inline since updates have no effect.

```suggestion
		const isInline = panels[0].type === 'inline';
		if (isInline) {
			panels.forEach((p, idx) => p._noBottomBorder = idx === panels.length - 1);
		}
```

Reply:
So the reason I didn't short-circuit like that (the old code did) is that it's possible that the panels start out as inline but then the type changes later, at which point we'd want to put the borders back.

Reply:
Wouldn't this run again then?

Reply:
We chatted offline and Danny's correct that we could short-circuit because there's no logic in the panels that relies on `_noBottomBorder` when the type isn't inline.

But we also agreed that it would be clearer if the reactive state property was just `_isLastPanelInGroup` and that's what the group was responsible for setting. Then the panel can do whatever it likes based on that, which for now will include hiding the bottom border when it's _not_ last and is inline. I'll update.